### PR TITLE
Add partialeq implementation for TryFromIntError type

### LINKED ISSUE #53458: PartialEq for TryFromIntError
Regarding code like this:

```
#![feature(try_from, never_type)]
use std::convert::TryFrom;
use std::num::TryFromIntError;

fn main() {
    let x: u8 = 125;
    let y: Result<u32, !> = u32::try_from(x);
    let _ = y == Ok(125); // OK

    let x: u32 = 125;
    let y: Result<u8, TryFromIntError> = u8::try_from(x);
    let _ = y.ok() == Some(125); // OK

    let x: u32 = 125;
    let y: Result<u8, TryFromIntError> = u8::try_from(x);
    let _ = y == Ok(125); // Error
}
```

I think I'd like the std library to add #[derive(PartialEq)] to std::num::TryFromIntError, so the last line compiles.

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
Fixes #53458.

### CODE REVIEW

File: src/test/run-pass/try-from-int-error-partial-eq.rs [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+#![feature(try_from)]
+
+use std::convert::TryFrom;
+use std::num::TryFromIntError;
+
+fn main() {
+    let x: u32 = 125;
+    let y: Result<u8, TryFromIntError> = u8::try_from(x);
+    let _ = y == Ok(125);
```
Comment (shepmaster) [inline src/test/run-pass/try-from-int-error-partial-eq.rs]: Can't this just be `y == Ok(125)`?

**Review by shepmaster — COMMENTED** [review]
My thoughts...

- I am a big fan of comparable error types for the reasons outlined in the issue. 

- This is unstable, so doesn't seem like a big risk

- We are already deriving `Copy` which feels like more of a forwards-compatibility hazard than `PartialEq`.

- But it's *still* a  forwards-compatibility hazard, so I don't feel qualified approving.

File: src/libcore/num/mod.rs [hunk 2]
```diff
@@ -4248,7 +4248,7 @@ from_str_radix_int_impl! { isize i8 i16 i32 i64 i128 usize u8 u16 u32 u64 u128 }
 
 /// The error type returned when a checked integral type conversion fails.
 #[unstable(feature = "try_from", issue = "33417")]
-#[derive(Debug, Copy, Clone)]
+#[derive(Debug, Copy, Clone, PartialEq)]
```
[the remark this answers is not in the document]
Comment (GuillaumeGomez) [reply src/libcore/num/mod.rs]: None indeed. Adding it.

**Comment by GuillaumeGomez** [comment]
Updated.

**Comment by KodrAus** [comment]
So as far as I know this error type is only attempting to communicate a conversion from one integral type to another where the value is outside of its possible range so I think from a forwards-compatibility perspective we're ok.

Thanks, this looks good to me @GuillaumeGomez!

@bors r+ rollup

**Pull request merged**